### DISCUSSION TITLE
Can’t connect HA to Modbus

### DISCUSSION THREAD

**Discussion by ssegruts** [nathanmarlor/foxess_modbus#663] [Support]
Followed the guides, hard wired cable from Inverter to Modbus using an EW11 and set a static IP.
Looking at the ip address, the device is showing serial traffic so I’m assuming the inverter and the wiring are all correct.
Getting the error message of: Error connecting to your adapter or inverter. Ensure the adapter is properly configured and is correctly wired to your inverter (see the setup link above), then try again. Details: Modbus Error: [Input/Output] timed out
Whatever I try, not getting any joy connecting.
Am I missing something obvious?

ta

**Comment by FozzieUK** [comment]
Have you been able to setup the EW11 as per the guide in https://github.com/nathanmarlor/foxess_modbus/wiki/Elfin-EW11 

The error implies there is no response from the device, but could well be the inverter - have you checked your RS485 cabling from inverter to the EW11, you should be using the 485A & B pins on the inverter (not the Meter485A/B) and they should be wired straight through so that A is wired to A on the EW11, and B to B

**Reply by ssegruts** [reply]
Yes, followed the guide 100%
From the attached screenshot, looks like the EW11 is receiving serial data but not passing it on.
It’s a KH7 
![IMG_0069](https://github.com/user-attachments/assets/8d43bb19-81f9-4f71-9190-232ecb55e487)

**Reply by FozzieUK** [reply]
What firmware versions is the KH running ?

**Comment by ssegruts** [comment]
1.29?

Pete
